Bottom: 17ad0806336425dff90f86d86787493fbe3eb088
Top:    514ea88f8631add7f29fd4864109d06c586ef744
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 23:09:45 +0100

Add a HID API call, notify_save_pcb() called around saving the PCB

The intention of this API is so that GUIs monitoring the active PCB
file on disk for changes, can filter out changes which occur as we
save the file ourselves.


---

diff --git a/src/file.c b/src/file.c
index 8021bcd..7c1ebc0 100644
--- a/src/file.c
+++ b/src/file.c
@@ -354,6 +354,9 @@ SavePCB (char *Filename)
   int retcode;
   char *copy;
 
+  if (gui->notify_save_pcb != NULL)
+    gui->notify_save_pcb (Filename, false);
+
   if (!(retcode = WritePipe (Filename, true)))
     {
       /* thanks to Nick Bailey for the bug-fix;
@@ -365,6 +368,10 @@ SavePCB (char *Filename)
       PCB->Filename = copy;
       SetChangedFlag (false);
     }
+
+  if (gui->notify_save_pcb != NULL)
+    gui->notify_save_pcb (Filename, true);
+
   return (retcode);
 }
 
diff --git a/src/hid.h b/src/hid.h
index b1f532f..d1c7c8e 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -567,6 +567,16 @@ typedef enum
      * Any remaining rendering will be flushed to the screen.
      */
     void (*finish_debug_draw)  (void);
+
+    /* Notification to the GUI around saving the PCB file.
+     *
+     * Called with a false parameter before the save, called again
+     * with true after the save.
+     *
+     * Allows GUIs which watch for file-changes on disk to ignore
+     * our deliberate changes.
+     */
+    void (*notify_save_pcb) (const char *filename, bool done);
   };
 
 /* Call this as soon as possible from main().  No other HID calls are
